tianshou/data/batch.py

```diff
@@ -263,18 +264,37 @@ def __init__(self,
                  **kwargs) -> None:
         if copy:
             batch_dict = deepcopy(batch_dict)
-        if _is_batch_set(batch_dict):
-            self.stack_(batch_dict)
-        elif isinstance(batch_dict, (dict, Batch)):
-            for k, v in batch_dict.items():
-                if isinstance(v, dict) or _is_batch_set(v):
-                    self.__dict__[k] = Batch(v)
-                else:
-                    if isinstance(v, list):
-                        v = np.array(v)
-                        if not issubclass(v.dtype.type, (np.bool_, np.number)):
-                            v = v.astype(np.object)
-                    self.__dict__[k] = v
+        if batch_dict is not None:
+            if isinstance(batch_dict, (dict, Batch)):
+                for k, v in batch_dict.items():
+                    if isinstance(v, (list, tuple, np.ndarray)):
+                        v_ = None
+                        if isinstance(v, np.ndarray):
```

Comment:
No need of this `if` case if you replace line 279 by `np.asanyarray` since it is free of cost when the argument is already an array.

Reply:
Good. I have changed the code.
```
a = np.asarray([1, 2, 3])
b = np.asanyarray(a)
b is a # True
c = np.array(a)
c is a # False
```

Reply:
Yes, that's the behavior I was referring to!